pymc/sampling/mcmc.py

```diff
@@ -304,7 +308,23 @@ def _sample_external_nuts(
                 "`var_names` are currently ignored by the nutpie sampler",
                 UserWarning,
             )
-        compiled_model = nutpie.compile_pymc_model(model)
+
+        def extract_backend(string: str) -> NutpieBackend:
+            match = re.search(r"(?<=\[)[^\]]+(?=\])", string)
+            if match is None:
+                return NUTPIE_DEFAULT_BACKEND
```

Comment:
You could also get a `None` match if the string is misformatted. For example, `nutpie[jax` would return a `None` match. I suggest that you test exact equality to set the default option, and if you get `None` then raise a `ValueError`.
```suggestion
            if string == "nutpie":
                return NUTPIE_DEFAULT_BACKEND
            elif match is None:
                raise ValueError(f"Could not parse nutpie backend. Found {string!r}")
```